app/models/consent_form_programme.rb

```diff
@@ -12,7 +12,6 @@
 #
 #  idx_on_programme_id_consent_form_id_2113cb7f37    (programme_id,consent_form_id) UNIQUE
 #  index_consent_form_programmes_on_consent_form_id  (consent_form_id)
-#  index_consent_form_programmes_on_programme_id     (programme_id)
```

Comment:
I'm curious to know here, if the index on `programme_id` is covered by the index on both `programme_id` and `consent_form_id`, why is the index only on `consent_form_id` not also covered by this index?

Reply:
Good question. I had to look this up myself. So apparently, in a composite index, the leftmost column can be used independently. This is known as the "leftmost prefix" rule. This is why `pg_hero` recommended removing the single-column index on `programme_id` - it's redundant. However, the same composite index does not cover queries that only use `consent_form_id`, because `consent_form_id` is not the leftmost column in the index.

Reply:
Ya, I remember this from my MySQL days. One way DBs implement composite indexes is simply by concating the values together, so you can see how only the first column can be user on it's own.

This also leads into how we might optimise an index by limiting it to be a partial index (only the first N chars) to balance out index size with lookup speed (for indexes that get too big to fit into RAM). We may have to look at this kind of thing in the future.